Bottom: 9cd6e354ee1a78b6fb489dd2fa6efb86074a70da
Top:    933efac75b2970b378a9362565c1c9539ee70d35
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2008-09-30 22:45:02 +0100

Add debugging code to visually check a pad's bounding box

Proved useful in fixing the pad bounding box update routine.
Disabled with #if 0 block.



---

diff --git a/src/draw.c b/src/draw.c
index c1d2784..96ab0b8 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -1475,6 +1475,16 @@ DrawPadLowLevel (hidGC gc, PadTypePtr Pad, Boolean clear, Boolean mask)
                       Pad->Point1.X, Pad->Point1.Y,
                       Pad->Point2.X, Pad->Point2.Y);
     }
+#if 0
+  { /* Draw bounding box for test */
+    BoxType *box = &Pad->BoundingBox;
+    gui->set_line_width (gc, 1);
+    gui->draw_line (gc, box->X1, box->Y1, box->X1, box->Y2);
+    gui->draw_line (gc, box->X1, box->Y2, box->X2, box->Y2);
+    gui->draw_line (gc, box->X2, box->Y2, box->X2, box->Y1);
+    gui->draw_line (gc, box->X2, box->Y1, box->X1, box->Y1);
+  }
+#endif
 }
 
 /* ---------------------------------------------------------------------------
